daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -167,6 +167,7 @@ object SValue {
 
     def apply(xs: Iterator[(SValue, SValue)]): SGenMap = {
       type O[_] = TreeMap[SValue, SValue]
+      xs.map { case p @ (k, _) => comparable(k); p }.to[O]
       SGenMap(xs.to[O])
```

Comment:
Sorry, I should have been clearer; you can't use the `xs` variable twice like this.

```suggestion
      SGenMap(xs.map { case p @ (k, _) => comparable(k); p }.to[O])
```

Reply:
You have been cleared. I was just planning to fix it latter. Thanks a lot anyway.